Work around gcc bug with ambiguous post/pre-increment.

Nicholas42 opened the pull request:
Works around https://github.com/joboccara/NamedType/issues/61 for arithmetic types.

MarekR22 reviewed (commented):
Respective test is messing.

Nicholas42 commented:
I added a test case and changed the way I did this while I was at it. I introduced `Incrementable` and `Decrementable` as skills, which should always be used if one wants to have both. Before, this was only fixed for `Arithmetic`. The test case from #61 will still not work, since there is no way around it, if we inherit from Post* and Pre* without adding the using directive.
